### PR TITLE
EventGrid: New preview API version with support for additional destination types, CloudEvents 1.0, and batching support.

### PR DESCRIPTION
**Pull request by kalyanaj**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [X] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [X] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [X] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by kalyanaj** [comment]
Commit 1 is an exact snapshot of the last preview API version (2019-02-01-preview) to make diffs easier. *Hence, the main changes are in commits 2, 3, 4*. Commit 5 is a change to merge the changes already done in our last GA version (2019-06-01) into this new preview API version.

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-01-01-preview/EventGrid.json [hunk 1]
```diff
@@ -2767,6 +2751,20 @@
           "description": "The base URL that represents the endpoint of the destination of an event subscription.",
           "type": "string",
           "readOnly": true
+        },
+        "maxEventsPerBatch": {
+          "format": "int32",
```
Comment (KrisBash) [inline specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-01-01-preview/EventGrid.json]: are there min/max values that could be defined in swagger?

**Review by KrisBash — APPROVED** [review]
LGTM

**Review by KrisBash — APPROVED** [review]
LGTM

**Comment by yungezz** [comment]
hi @ruowan could you pls help to look at what's the avocado error?

**Comment by ruowan** [comment]
Hi, @kalyanaj  For avocado validation error. You need to add `/specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-01-01-preview/EventGrid.json` to your readme file as an input file. Because if the readme doesn't contain this file, then SDK will not be generated and avocado report UNREFERENCED_JSON_FILE error.

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-01-01-preview/EventGrid.json (hunk 1 above)
Comment (kalyanaj) [reply specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-01-01-preview/EventGrid.json]: Thanks for the review!  Currently the limits are enforced by our RP.  Once we close on the exact values, we will add the min/max values in swagger for our stable version.

**Comment by kalyanaj** [comment]
Thanks @ruowan for the info!  I have updated the README files now and CI is green now. Please let me know if you have any additional feedback Can you please help review/merge this PR?

**Pull request merged**